src/containerapp/azext_containerapp/containerapp_session_code_interpreter_decorator.py

```diff
@@ -0,0 +1,159 @@
+        self.session_code_interpreter_def = SessionPoolModel
+
+    def get_argument_name(self):
+        return self.get_param('name')
+
+    def get_argument_resource_group_name(self):
+        return self.get_param('resource_group_name')
+
+    def get_argument_identifier(self):
+        return self.get_param('identifier')
+    
+    def get_argument_code(self):
+        return self.get_param('code')
+    
+    def get_argument_timeout_in_seconds(self):
+        return self.get_param('timeout_in_seconds')
+        
+    def get_argument_filename(self):
+        return self.get_param('filename')
+    
+    def get_argument_filepath(self):
+        return self.get_param('filepath')    
+        
+    def get_argument_path(self):
+        return self.get_param('path')    
+          
+    def get_argument_sessionpool_location(self):
+        return self.get_param('session_pool_location')    
+        
+    def get_sessionpool_endpoint(self):
+        # if the user provides the session pool location get session pool endpoint by 
+        # constructing the url endpoint ourselves so command runs faster
+        # otherwise get the session pool endpoint by doing a get on the session pool
+        session_pool_endpoint = None
+        if self.get_argument_sessionpool_location() is not None:
+            session_pool_endpoint_fmt = "https://{}.dynamicsessions.io/subscriptions/{}/resourceGroups/{}/sessionPools/{}"
+            session_pool_endpoint = session_pool_endpoint_fmt.format(
+                self.get_argument_sessionpool_location(),
+                get_subscription_id(self.cmd.cli_ctx),
+                self.get_argument_resource_group_name() ,
+                self.get_argument_name())
+        else:
+            sessionpool =  SessionPoolPreviewClient.show(cmd=self.cmd, resource_group_name=self.get_argument_resource_group_name(),
+                    name=self.get_argument_name())
+            session_pool_endpoint = sessionpool["properties"]["poolManagementEndpoint"]
+        return session_pool_endpoint
+
+class SessionCodeInterpreterCommandsPreviewDecorator(SessionCodeInterpreterPreviewDecorator):
+    def validate_arguments(self):
+        if self.get_argument_sessionpool_location() is not None:
+           _ensure_location_allowed(self.cmd, self.get_argument_sessionpool_location(), CONTAINER_APPS_RP, "sessionPools")
+
+    def construct_payload(self):
+        self.session_code_interpreter_def["properties"]["identifier"] = self.get_argument_identifier()
+
+        self.session_code_interpreter_def["properties"]["codeInputType"] = "inline"
+        self.session_code_interpreter_def["properties"]["executionType"] = "synchronous"
+        
+        if self.get_argument_timeout_in_seconds() is not None:
+            self.session_code_interpreter_def["properties"]["timeoutInSeconds"] = self.get_argument_timeout_in_seconds()
+        else :
+            self.session_code_interpreter_def["properties"]["timeoutInSeconds"] = 60
```

Comment:
So the default value of `timeout_in_seconds` is 60, right?
You should add the default in the params directly, make the customer can know what the default value is.

Reply:
sounds good! added that and fixed